modules/internal/ChapelBase.chpl

```diff
@@ -2012,6 +2012,7 @@ module ChapelBase {
         halt("Attempt to compute a modulus by zero");
     __primitive("%=", lhs, rhs);
   }
+  deprecated "'%=' is deprecated for 'real' values with no expectation to replace it"
```

Comment:
Do other modern languages support `%` on reals. We do have a `mod` for reals (and c has `fmod`), so seems like this could be a convenience wrapper for that? I don't think that's something we need to figure out today, but if we might want to support in the future I guess I'd drop the "with no expectation to replace it"

Reply:
Right, and the spec section https://chapel-lang.org/docs/language/spec/expressions.html#modulus-operators ends by stating an expectation to support it (at some point)

Reply:
I'd forgotten about this intent until Damian reminded me about it last night, but there are a few issues on this topic:
* #19255
* #6205
Here, I'm just trying to pull support for something that seems obviously broken and not ready for users rather than fulfilling those issues.  I'll mention them in the OP now that I've got them back in cache, though.

Thanks for the explanation about the change in behavior, Michael!  I had definitely not noticed the where clause change when looking last night.

Reply:
(and good point about rewording the deprecation message based on that, Elliot).